plugins/kubernetes-backend/src/kubernetes-auth-translator/KubernetesAuthTranslatorGenerator.test.ts

```diff
@@ -14,46 +14,78 @@
  * limitations under the License.
  */
 
-import { KubernetesAuthTranslator } from './types';
-import { GoogleKubernetesAuthTranslator } from './GoogleKubernetesAuthTranslator';
 import { KubernetesAuthTranslatorGenerator } from './KubernetesAuthTranslatorGenerator';
-import { NoopKubernetesAuthTranslator } from './NoopKubernetesAuthTranslator';
-import { AwsIamKubernetesAuthTranslator } from './AwsIamKubernetesAuthTranslator';
-import { OidcKubernetesAuthTranslator } from './OidcKubernetesAuthTranslator';
-import { getVoidLogger } from '@backstage/backend-common';
+import { ClusterDetails } from '../types';
+import { KubernetesRequestAuth } from '@backstage/plugin-kubernetes-common';
 
-const logger = getVoidLogger();
+const mockTranslator = jest.fn();
 
-describe('getKubernetesAuthTranslatorInstance', () => {
-  const sut = KubernetesAuthTranslatorGenerator;
+const authTranslatorMap = {
+  google: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  aws: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  azure: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  serviceAccount: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  googleServiceAccount: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  oidc: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  localKubectlProxy: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+};
 
-  it('can return an auth translator for google auth', () => {
-    const authTranslator: KubernetesAuthTranslator =
-      sut.getKubernetesAuthTranslatorInstance('google', { logger });
-    expect(authTranslator instanceof GoogleKubernetesAuthTranslator).toBe(true);
-  });
+describe('decorateClusterDetailsWithAuth', () => {
+  let authTranslator: KubernetesAuthTranslatorGenerator;
+  const authObject: KubernetesRequestAuth = {};
 
-  it('can return an auth translator for aws auth', () => {
-    const authTranslator: KubernetesAuthTranslator =
-      sut.getKubernetesAuthTranslatorInstance('aws', { logger });
-    expect(authTranslator instanceof AwsIamKubernetesAuthTranslator).toBe(true);
+  beforeEach(() => {
+    mockTranslator.mockClear();
+    authTranslator = new KubernetesAuthTranslatorGenerator({
+      authTranslatorMap,
+    });
   });
 
-  it('can return an auth translator for serviceAccount auth', () => {
-    const authTranslator: KubernetesAuthTranslator =
-      sut.getKubernetesAuthTranslatorInstance('serviceAccount', { logger });
-    expect(authTranslator instanceof NoopKubernetesAuthTranslator).toBe(true);
-  });
+  it('can decorate cluster details if the auth provider is in the translator map', () => {
+    const expectedClusterDetails: ClusterDetails = {
+      url: 'notanything.com',
+      name: 'randomName',
+      authProvider: 'google',
+    };
```

Comment:
Maybe something here to prove that the generator doesn't just return an un-decorated `ClusterDetails`?
```suggestion
    const expectedClusterDetails: ClusterDetails = {
      url: 'notanything.com',
      name: 'randomName',
      authProvider: 'google',
      serviceAccountToken: 'added by mock translator',
    };
```

Reply:
Done